### PR TITLE
using less misleading message in case all DNS resolvers fail

### LINKED ISSUE #26097: DNS errors are always reported as "Timed out resolving"
The AsyncDnsResolver tries all available resolvers, then returns `ResolveFailedException(s"Timed out resolving $name with nameservers: $nameServers")` regardless of whether the error was actually a timeout. It would be good to use a less misleading message.

### PR DESCRIPTION
**Pull request by mszczygiel**
resolves https://github.com/akka/akka/issues/26097

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala [hunk 1]
```diff
@@ -89,6 +89,20 @@ class AsyncDnsResolverSpec extends AkkaSpec(
...
+      val dnsClient1 = TestProbe()
+      val dnsClient2 = TestProbe()
+      val r = resolver(List(dnsClient1.ref, dnsClient2.ref))
+      r ! Resolve("cats.com", Ip(ipv4 = true, ipv6 = false))
+      dnsClient1.expectMsg(Question4(1, "cats.com"))
+      dnsClient1.reply(Failure(new RuntimeException("Fail")))
+      dnsClient2.expectMsg(Question4(2, "cats.com"))
+      dnsClient2.reply(Failure(new RuntimeException("Yet another fail")))
+      expectMsgPF(remainingOrDefault) {
+        case Failure(ResolveFailedException(_)) ⇒
```
Comment (raboof) [inline akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala]: Should we check the message here?

**Review by raboof — APPROVED** [review]
Nice thanks!

**Comment by raboof** [comment]
OK TO TEST

File: akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala (hunk 1 above)
Comment (mszczygiel) [reply akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala]: I think there is no much value checking it - `ResolveFailedException` is created with only one message.

**Review by johanandren — APPROVED** [review]

**Pull request merged**